### PR TITLE
19453 - Amalgamation numbered change

### PR DESCRIPTION
**Pull request by seeker25**

### CODE REVIEW

File: legal-api/src/legal_api/resources/v2/business/business.py [hunk 1]
```diff
@@ -167,19 +167,24 @@ def search_businesses():
         draft_query = db.session.query(Filing).filter(
```
Comment (seeker25) [inline legal-api/src/legal_api/resources/v2/business/business.py:167]: Would be nice to put these in a service, but just trying to do smallest changes possible here

File: legal-api/src/legal_api/resources/v2/business/business.py [hunk 2]
```diff
@@ -167,19 +167,24 @@ def search_businesses():
...
+            }
+            if draft_dao.json_nr:
+                draft['nrNumber'] = draft_dao.json_nr
+            draft['legalName'] = (draft_dao.filing_json.get('filing', {})
+                                        .get(draft_dao.filing_type, {})
+                                        .get('nameRequest', {})
+                                        .get('legalName'))
+            draft['draftType'] = Filing.FILINGS.get(draft_dao.filing_type, {}).get('temporaryCorpTypeCode')
+            if draft['draftType'] == 'ATMP' and draft['legalName'] is None:
+                draft['legalName'] = 'Numbered Amalgamated Company'
```
Comment (seeker25) [inline legal-api/src/legal_api/resources/v2/business/business.py:186]: Alternative to this, would be doing this:
https://github.com/bcgov/sbc-auth/pull/2705

**Review by PaulGarewal — APPROVED** [review]
LGTM 👍

**Comment by seeker25** [comment]
Scenario 1 NR:

image

Scenario 2 no NR with legalName: 

image

Scenario 3 no NR, no legalName:

image

File: legal-api/src/legal_api/resources/v2/business/business.py [hunk 3]
```diff
@@ -167,19 +167,24 @@ def search_businesses():
...
+                'legalType': draft_dao.json_legal_type
+            }
+            if draft_dao.json_nr:
+                draft['nrNumber'] = draft_dao.json_nr
+            draft['legalName'] = (draft_dao.filing_json.get('filing', {})
+                                        .get(draft_dao.filing_type, {})
+                                        .get('nameRequest', {})
+                                        .get('legalName'))
+            draft['draftType'] = Filing.FILINGS.get(draft_dao.filing_type, {}).get('temporaryCorpTypeCode')
+            if draft['draftType'] == 'ATMP' and draft['legalName'] is None:
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/business/business.py:185]: draft['legalName'] can be None for Incorporation Application (if its numbered)

Comment (seeker25) [reply legal-api/src/legal_api/resources/v2/business/business.py:185]: will fix

**Review by vysakh-menon-aot — APPROVED** [review]

File: legal-api/src/legal_api/resources/v2/business/business.py (hunk 3 above)
Comment (seeker25) [reply legal-api/src/legal_api/resources/v2/business/business.py:185]: Can IA have draftType = ATMP?

Comment (seeker25) [reply legal-api/src/legal_api/resources/v2/business/business.py:185]: I see, we could put in for IA as well.. hmm one sec

Comment (seeker25) [reply legal-api/src/legal_api/resources/v2/business/business.py:185]: I think we'd have to put most of this code in here:

https://github.com/bcgov/bcrs-shared-components/blob/main/src/modules/corp-type-module/corp-type-module.ts
To compute IA's numbered text
```
            if draft['draftType'] == 'TMP' and not draft_dao.json_nr and draft_dao.json_legal_type in ('BC', 'BEN','ULC', 'CCC'):
                draft['legalName'] = generate_numbered_ia_name()
```

Comment (seeker25) [reply legal-api/src/legal_api/resources/v2/business/business.py:185]:
``` 
draft_results = []
        ia_numbered_dict = {
            'BC': 'Numbered Limited Company',
            'BEN': 'Numbered Benefit Company',
            'ULC': 'Numbered Unlimited Liability Company',
            'CCC': 'Numbered Community Contribution Company'
        }
        for draft_dao in draft_query.all():
            draft = {
                'identifier': draft_dao.temp_reg,  
                'legalType': draft_dao.json_legal_type
            }
            if draft_dao.json_nr:
                draft['nrNumber'] = draft_dao.json_nr
            draft['legalName'] = (draft_dao.filing_json.get('filing', {})
                                        .get(draft_dao.filing_type, {})
                                        .get('nameRequest', {})
                                        .get('legalName'))
            draft['draftType'] = Filing.FILINGS.get(draft_dao.filing_type, {}).get('temporaryCorpTypeCode')
            if draft['draftType'] == 'ATMP' and draft['legalName'] is None:
                draft['legalName'] = 'Numbered Amalgamated Company'
            if draft['draftType'] == 'TMP' and not draft_dao.json_nr:
                draft['legalName'] = ia_numbered_dict.get(draft_dao.json_legal_type, None)
            draft_results.append(draft)
```

Comment (seeker25) [reply legal-api/src/legal_api/resources/v2/business/business.py:185]: something like that?

Comment (seeker25) [reply legal-api/src/legal_api/resources/v2/business/business.py:185]:
```
    BUSINESSES = {
        LegalTypes.BCOMP: {
            'numberedLegalNameSuffix': 'B.C. LTD.',
            'numberedDescription': 'Numbered Benefit Company'
        },
        LegalTypes.COMP: {
            'numberedLegalNameSuffix': 'B.C. LTD.',
            'numberedDescription': 'Numbered Limited Company'
        },
        LegalTypes.BC_ULC_COMPANY: {
            'numberedLegalNameSuffix': 'B.C. UNLIMITED LIABILITY COMPANY',
            'numberedDescription': 'Numbered Unlimited Liability Company'
        },
        LegalTypes.BC_CCC: {
            'numberedLegalNameSuffix': 'B.C. COMMUNITY CONTRIBUTION COMPANY LTD.',
            'numberedDescription': 'Numbered Community Contribution Company'
        }
    }
    ```

**Comment by seeker25** [comment]
@vysakh-menon-aot I'm going to merge this one, I'll look at adding in the TMP another time

**Pull request merged**